src/server/obj-info.c

```diff
@@ -67,8 +67,69 @@ static cptr act_description[ACT_MAX] =
 	"berserk rage (50+d50 turns)"
 };
 
+/* MAngband specific. Math sourced from cmd6.c. 
+ * Give the player an idea of their odds of successfully activating 
+ * a wand, staff, rod, or item. - Avenger */
+void describe_activation_chance(const object_type *o_ptr)
+{
+	int	chance, lev, maskIndex;
+	char *mask[10] = {
+		"almost no",
+		"a negligible",
+		"a very low",
+		"a low",
+		"a moderate",
+		"a good",
+		"a very good",
+		"an excellent",
+		"an almost certain",
+	};
+	player_type *p_ptr = Players[player_textout];
```

Comment:
The use of global/static variables in functions like this is concerning. Is this safe?

Reply:
I've updated this to remove the declaration and simply branch on the value of chance with a complete message of the form text_out("\n\n   You have [foo] chance to activate this item"). This is in line with how every other output message in the class is treated - if there is a better way to do this, it is not evident in obj-info.c.